platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ConfigurationBuildDependenciesIntegrationTest.groovy

```diff
@@ -43,7 +43,7 @@ class ConfigurationBuildDependenciesIntegrationTest extends AbstractHttpDependen
                 compile files('main-lib.jar') { builtBy lib }
             }
             task direct { inputs.files configurations.compile }
-            task fileCollection { inputs.files configurations.compile.fileCollection { true } }
+            task artifactView { inputs.files configurations.compile.incoming.artifactView { componentFilter { true } }.files }
```

Comment:
🤔 The `componentFilter { true }` doesn't seem to be needed here, this applies elsewhere in the test changes for this PR.

Reply:
Many of the tests here use `componentFilter { true }` as one of several test cases, used to ensure that certain test expectations hold regardless of the way a `FileCollection` is acquired from the API. 

Extra logic _does_ execute when a componentFilter is provided. I think it is reasonable to cover those cases